smart-contracts/wasm-chain-integration/src/v1/ffi.rs

```diff
@@ -202,7 +201,8 @@ unsafe extern "C" fn call_init_v1(
 #[no_mangle]
 unsafe extern "C" fn call_receive_v1(
     loader: LoadCallback,
```

Comment:
The doc comment of this function still talks about artifact pointers. Please update and explain the new ownership design.